intake_xarray/opendap.py

```diff
@@ -15,25 +15,44 @@ class OpenDapSource(DataSourceMixin):
         Chunks is used to load the new dataset into dask
         arrays. ``chunks={}`` loads the dataset with dask using a single
         chunk for all arrays.
+    auth: None, "esgf" or "urs"
+        Method of authenticating to the OPeNDAP server.
+        Choose from one of the following:
+        'esgf' - [Default] Earth System Grid Federation.
+        'urs' - NASA Earthdata Login, also known as URS.
+        None - No authentication.
+        Note that you will need to set your username and password respectively using the
+        environment variables DAP_USER and DAP_PASSWORD.
     """
     name = 'opendap'
 
-    def __init__(self, urlpath, chunks, xarray_kwargs=None, metadata=None,
+    def __init__(self, urlpath, chunks, auth="esgf", xarray_kwargs=None, metadata=None,
```

Comment:
Are you sure you want the default auth to be `esgf`?

Reply:
[`ESGF`](https://www.pydap.org/en/latest/client.html#earth-system-grid-federation-esgf) was the 'default' authentication method in the original code. Personally I would prefer using `URS` as the default, but might be good to preserve backward compatibility here.

Reply:
Defaults are hard. Perhaps, @danielballan or @jsignell care to weigh in.

Reply:
I'm good with @martindurant's comment in the main thread. :+1:

Reply:
Yep. Same for me